src/Analyzer/CodeAnalyzer.cs

```diff
@@ -0,0 +1,154 @@
+using TSQLLint.Infrastructure.Reporters;
+using Codacy.Engine.Seed;
+using Codacy.Engine.Seed.Results;
+using Codacy.TSQLLint.Configuration;
+using Codacy.TSQLLint.Reporters;
+using Newtonsoft.Json;
+
+namespace Codacy.TSQLLint
+{
+    /// <summary>
+    /// Tool integration for TSQLLint using the Seed class.
+    /// </summary>
+    public class CodeAnalyzer : Engine.Seed.CodeAnalyzer
+    {
+        private readonly IConfigReader configReader;
+        private readonly IReporter reporter;
+        private readonly IConsoleTimer timer;
+
+        private const string sqlExtension = ".sql";
+        private const string defaultTSQLLintConfiguration = ".tsqllintrc";
+
+        private readonly string tmpTSQLLintPath;
+
+        /// <summary>
+        /// Tool integration constructor.
+        /// This will prepare everything needed for the tool to work with codacy
+        /// integration, using the Seed.
+        /// </summary>
+        public CodeAnalyzer() : base(sqlExtension)
+        {
+            this.timer = new ConsoleTimer();
+            this.reporter = new CodacyConsoleReporter();
+            this.configReader = new ConfigReader(reporter);
+
+            // create temporary directory
+            var tmpTSQLLintFolder = Path.Combine(Path.GetTempPath(), "tsqllint_" + Guid.NewGuid());
+            Directory.CreateDirectory(tmpTSQLLintFolder);
+
+            this.tmpTSQLLintPath = Path.Combine(tmpTSQLLintFolder, defaultTSQLLintConfiguration);
+
+            var tsqllintConfig = new TSQLLintConfiguration();
+
+            if (!(PatternIds is null) && PatternIds.Any())
+            {
+                tsqllintConfig.Rules = new Dictionary<string, string>();
+                foreach (var pattern in CurrentTool.Patterns)
+                {
+                    tsqllintConfig.Rules.Add(pattern.PatternId, "error");
+                }
+
+                foreach (var unusedPattern in Patterns.Patterns.Select(p => p.PatternId)
+                    .Except(CurrentTool.Patterns.Select(p => p.PatternId)))
+                {
+                    tsqllintConfig.Rules.Add(unusedPattern, "off");
+                }
+            } else if (File.Exists(defaultTSQLLintConfiguration))
+            {
+                var tsqlliteJSON = File.ReadAllText(defaultTSQLLintConfiguration);
+                var currentTSQLLintConfig = JsonConvert.DeserializeObject<TSQLLintConfiguration>(tsqlliteJSON);
+                tsqllintConfig.Rules = currentTSQLLintConfig.Rules;
+                tsqllintConfig.CompatibilityLevel = currentTSQLLintConfig.CompatibilityLevel;
+            }
+            else
+            {
+                tsqllintConfig.Rules = new Dictionary<string, string>();
+                foreach (var pattern in Patterns.Patterns.Select(p => p.PatternId))
+                {
+                    tsqllintConfig.Rules.Add(pattern, "error");
+                }
+            }
+
+            if (tsqllintConfig.CompatibilityLevel is null)

```

Comment:
I was told by @gjsduarte that it does exists an amazing default library for loading configuration and managing fallback.
Can you check and avoid these manual switches?

Reply:
IDK what do you mean by that, can you show me an example ?

This is the only check I'm doing for a default fallback in the configuration files.

The other checks are to manipulate the config file to use codacy patterns and make the expected behavior of a codacy tool like we have on all tools. I'm following the same logic as scala seed documentation.

Reply:
since it's just one I don't mind